src/wrapper.cpp

```diff
@@ -42,25 +54,36 @@ PYBIND11_MODULE(pyleopart, m)
            py::overload_cast<>(&Particles::Particles::cell_particles,
                                py::const_));
 
+  // Projection classes
+  py::class_<L2Project>(m, "L2Project")
+      .def(py::init<Particles&,
+                    std::shared_ptr<dolfinx::function::Function<PetscScalar>>,
+                    std::string>())
+      .def("solve", py::overload_cast<>(&L2Project::L2Project::solve),
+           "l2 projection")
+      .def("solve",
+           py::overload_cast<double, double>(&L2Project::L2Project::solve),
+           "Bounded l2 projection");
+
   // Generation functions
-  m.def("random_tet", &generation::random_reference_tetrahedron);
-  m.def("random_tri", &generation::random_reference_triangle);
-  m.def("mesh_fill", &generation::mesh_fill);
+  m.def("random_tet", &random_reference_tetrahedron);
+  m.def("random_tri", &random_reference_triangle);
+  m.def("mesh_fill", &mesh_fill);
 
-  // Transfer functions
-  m.def("get_particle_contributions", &transfer::get_particle_contributions);
+  // Transfer functions (probably shouldn't be interfaced)
```

Comment:
Maybe OK for testing in pytest. They might not be in the "public" interface.

Reply:
Not sure how we can decide "after" the wrapper whether something should be publicly interfaced or not?

Reply:
One option would be to give them a name starting with "_" which is public, but showing that it is not really for general use.